packages/server/tests/integration/v1/auth/password.spec.ts

```diff
@@ -0,0 +1,192 @@
+import { describe, expect, it } from "vitest";
+import supertest from "supertest";
+import { faker } from "@faker-js/faker";
+import jwt from "jsonwebtoken";
+
+import app from "../../../../src/app";
+import { createToken } from "../../../../src/services/token.service";
+import database from "../../../../src/database";
+import { createUser } from "../../../utils/seed/user";
+
+describe("POST /api/v1/auth/password/reset", () => {
+  it('should throw error "EMAIL_INVALID" when invalid email is sent', async () => {
+    const response = await supertest(app)
+      .post("/api/v1/auth/password/reset")
+      .send({ email: "not-an-email" });
+
+    expect(response.headers["content-type"]).toContain("application/json");
+    expect(response.status).toBe(400);
+    expect(response.body.code).toBe("EMAIL_INVALID");
+  });
+
+  it('should throw error "USER_NOT_FOUND" when email does not exist', async () => {
+    const response = await supertest(app)
+      .post("/api/v1/auth/password/reset")
+      .send({ email: faker.internet.email() });
+
+    expect(response.headers["content-type"]).toContain("application/json");
+    expect(response.status).toBe(404);
+    expect(response.body.code).toBe("USER_NOT_FOUND");
+  });
+
+  it("should send password reset mail and return token (200)", async () => {
+    const { user } = await createUser();
+
+    const response = await supertest(app)
+      .post("/api/v1/auth/password/reset")
+      .send({ email: user.email });
+
+    expect(response.status).toBe(200);
+    expect(response.headers["content-type"]).toContain("application/json");
+  });
+});
+
+describe("POST /api/v1/auth/password/validateToken", () => {
+  it("should throw error MISSING_TOKEN if token is missing", async () => {
+    const response = await supertest(app).post(
+      "/api/v1/auth/password/validateToken",
+    );
+
+    expect(response.headers["content-type"]).toContain("application/json");
+    expect(response.status).toBe(400);
+    expect(response.body.errors[0].code).toBe("MISSING_TOKEN");
```

Comment:
Instead of hard-coding the array index, can you try using `expect.arrayContaining()`